### DISCUSSION TITLE
Running N independent version of a job

### DISCUSSION THREAD

**Discussion by smilenstoychev** [dagster-io/dagster#15797] [Q&A]
I'm looking for a way to have a job that can be run multiple times for a combination of unique config parameters. 

For example: If there are N users with their own individual data (all of a fixed shape) I want a job run for each user to be uniquely identifiable and independent from the job run of a different user. 

If I start a job with **user1** parameter, that job would only works on assets in a predefined location corresponding to **user1** and each time it is re-materialized it will only affect the **user1** job run. A job run for **user2** would not affect it's staleness or run status, essentially it will be a completely separate independent job, which only shares a definition. Is there a straighforward way to achieve this?

**Comment by skeller88** [comment]
Seems like you're looking for a lot of the features in [branch deployments](https://docs.dagster.io/dagster-cloud/managing-deployments/branch-deployments#branch-deployments-in-dagster-cloud)? You could also try partitioning assets by username.
